components/brave_wallet/browser/meld_integration_service.cc

```diff
@@ -535,4 +726,172 @@ void MeldIntegrationService::OnParseCountries(
+      FROM_HERE,
+      base::BindOnce(&ParseCryptoWidgetCreate, api_request_result.TakeBody()),
+      base::BindOnce(&MeldIntegrationService::OnParseCryptoBuyWidgetCreate,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void MeldIntegrationService::CryptoSellWidgetCreate(
+    mojom::CryptoSellSessionDataPtr session_data,
+    mojom::CryptoWidgetCustomerDataPtr customer_data,
+    CryptoSellWidgetCreateCallback callback) {
+  auto internal_callback =
+      base::BindOnce(&MeldIntegrationService::OnCryptoSellWidgetCreate,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  auto payload_value = GetCryptoSellWidgetPayload(std::move(session_data),
+                                                  std::move(customer_data));
+
+  auto url = GURL(kMeldRpcEndpoint).Resolve("/crypto/session/widget");
+  auto conversion_callback = base::BindOnce(&SanitizeJson);
+  api_request_helper_->Request(
+      "POST", url, GetPayload(payload_value), "", std::move(internal_callback),
+      MakeMeldApiHeaders(), {.auto_retry_on_network_change = true},
+      std::move(conversion_callback));
+}
+
+void MeldIntegrationService::OnParseCryptoSellWidgetCreate(
+    CryptoSellWidgetCreateCallback callback,
+    mojom::MeldCryptoWidgetPtr crypto_widget) const {
+  if (!crypto_widget) {
+    std::move(callback).Run(
+        nullptr, std::vector<std::string>{
+                     l10n_util::GetStringUTF8(IDS_WALLET_PARSING_ERROR)});
+    return;
+  }
+
+  std::move(callback).Run(std::move(crypto_widget), std::nullopt);
+}
+
+void MeldIntegrationService::OnCryptoSellWidgetCreate(
+    CryptoSellWidgetCreateCallback callback,
+    APIRequestResult api_request_result) const {
+  if (!api_request_result.Is2XXResponseCode() &&
+      !NeedsToParseResponse(api_request_result.response_code())) {
+    std::move(callback).Run(
+        nullptr, std::vector<std::string>{
+                     l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR)});
+    return;
+  }
+
+  if (auto errors = ParseMeldErrorResponse(api_request_result.value_body());
+      errors) {
+    std::move(callback).Run(nullptr, errors);
+    return;
+  }
+
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE,
+      base::BindOnce(&ParseCryptoWidgetCreate, api_request_result.TakeBody()),
+      base::BindOnce(&MeldIntegrationService::OnParseCryptoSellWidgetCreate,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void MeldIntegrationService::CryptoTransferWidgetCreate(
+    mojom::CryptoTransferSessionDataPtr session_data,
+    mojom::CryptoWidgetCustomerDataPtr customer_data,
+    CryptoTransferWidgetCreateCallback callback) {
+  auto internal_callback =
+      base::BindOnce(&MeldIntegrationService::OnCryptoSellWidgetCreate,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  auto payload_value = GetCryptoTransferWidgetPayload(std::move(session_data),
```

Comment:
Should we still execute api call if payload_value is nullopt or we should threat this as an error case?